frontend/appflowy_flutter/lib/plugins/database/grid/presentation/mobile_grid_page.dart

```diff
@@ -90,19 +102,43 @@ class _MobileGridPageState extends State<MobileGridPage> {
           return state.loadingState.map(
             loading: (_) =>
                 const Center(child: CircularProgressIndicator.adaptive()),
-            finish: (result) => result.successOrFail.fold(
-              (_) => GridShortcuts(child: GridPageContent(view: widget.view)),
-              (err) => FlowyErrorPage.message(
-                err.toString(),
-                howToFix: LocaleKeys.errorDialog_howToFixFallback.tr(),
-              ),
-            ),
+            finish: (result) {
+              _openRow(context, widget.initialRowId, true);
```

Comment:
perform logic here and early return when possible. Reduce responsibility of _openRow to just opening the row.